### ISSUE TITLE
How to ensure the LRU cache is thread safe?

### ISSUE THREAD

**Issue by uran0sH** [dermesser/leveldb-rs#22]
The LRU cache implemented in cache.rs is not thread safe. I'd like to know how you ensure that it is thread-safe

**Comment by dermesser** [comment]
Well as you say, it isn't thread-safe and not made to be. That's why it isn't `Sync`.

Do you mean to ask how it could be made thread-safe?

**Comment by uran0sH** [comment]
Yeah. I thought you made sure the LRU cache was thread-safe elsewhere. I wrote a perhaps thread-safe LRU cache. Maybe you can give some advice? https://github.com/uran0sH/dsa-rs/blob/main/src/concurrent_lru.rs

**Comment by dermesser** [comment]
As my cache isn't concurrent, it is built quite differently from yours, so right away I can't give good advice I'm afraid.
